### PR TITLE
Add junos_system declartive module and other related change

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

*  junos_system declartive module
*  integration test for junos_system
*  integration test for net_system (junos platform)
*  pep8 fixes for junos modules
*  move to lxml from elementree for xml parsing as it support
   complete set of xpath api's
*  other minor changes

##### ISSUE TYPE

 - Feature Pull Request
 - Docs Pull Request

##### COMPONENT NAME

junos_system
junos_command
junos_config
junos_package
junos_rpc
junos_user
junos_vlan
junos_interface

##### ANSIBLE VERSION

```
2.4
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by gundalow** [comment]
Unit tests are failing as lxml isn't installed

File: lib/ansible/module_utils/junos.py [hunk 1]
```diff
@@ -17,16 +17,22 @@
...
 from ansible.module_utils.netconf import send_request, children
 from ansible.module_utils.netconf import discard_changes, validate
 from ansible.module_utils.six import string_types
 from ansible.module_utils._text import to_text
 
+try:
+    from lxml.etree import Element, SubElement, fromstring, tostring
+    HAS_LXML = True
+except ImportError:
+    from xml.etree.ElementTree import SubElement, fromstring, tostring
```
Comment (Qalthos) [inline lib/ansible/module_utils/junos.py]: You appear to have dropped `Element`

File: lib/ansible/module_utils/netconf.py [hunk 2]
```diff
@@ -26,10 +26,14 @@
...
 #
 from contextlib import contextmanager
-from xml.etree.ElementTree import Element, SubElement, fromstring, tostring
 
 from ansible.module_utils.connection import exec_command
 
+try:
+    from lxml.etree import Element, SubElement, fromstring, tostring
+except ImportError:
+    from xml.etree.ElementTree import SubElement, fromstring, tostring
```
Comment (Qalthos) [inline lib/ansible/module_utils/netconf.py]: Again, no `Element` here?

**Review by Qalthos — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/junos.py (hunk 1 above)
Comment (ganeshrn) [reply lib/ansible/module_utils/junos.py]: Fixed

**Review by Qalthos — APPROVED** [review]

File: lib/ansible/module_utils/netconf.py (hunk 2 above)
Comment (ganeshrn) [reply lib/ansible/module_utils/netconf.py]: Fixed

**Comment by ganeshrn** [comment]
@Qalthos Thank you for review

**Pull request merged**